### ISSUE TITLE
[WCAG 2.2 Draft Feedback] Success Criterion 2.4.12 Focus Appearance (Enhanced)

### ISSUE THREAD

**Issue by dshoukry** [w3c/wcag#1451]
“Success Criterion 2.4.12 Focus Appearance (Enhanced) (Level AAA): For the keyboard focus indicator of each User Interface Component, all of the following are true:

- Minimum area: The focus indication area is greater than or equal to a 2 CSS pixel solid border around the control.
- Change of contrast: Color changes used to indicate focus have a contrast ratio of at least 4.5:1 with the colors changed from the unfocused control.
- Unobscured: No part of the focus indicator is hidden by author-created content.”

Our concerns are mostly the same as those raised in the corresponding AA level requirement. Can the group explain the decision-making process that influenced the “step up” between the AA and AAA requirement and confirm which (if any) of our AA feedback will be accepted and carried over into the AAA requirement?

Please find detailed specifics covered in our [Success Criterion 2.4.12 Focus Appearance (Enhanced) (AAA) Google Doc](https://docs.google.com/document/d/1kvYQbwbdYmgFxrIT8ph6bUWUSIVcg-RFAQET3przPwE/edit#heading=h.ks4520ui4r6v).

**Comment by alastc** [comment]
Draft response:
<hr>

The aspects raised on the AA version will be dealt with in that thread, noting that it has changed and it is likely you would want to re-review it.

On the specific points raised, with the 'change of contrast' bullet highlighted you asked:

> Should the contrast ratio be tied to object size? If you apply zoom to the page the contrast ratio might change, and the nearest surrounding controls/colors may also change as well.

The size requirement is separate, and is what the 1st bullet is for.

Zoom may influence which colours are adjacent, but each 'page variation' should be tested. 

I'm not sure what value would be gained by the contrast ratio being tied to the object size? We had considered having 4.5:1 for 1px outlines, and 3:1 for thicker outlines, but that didn't seem to work outside of a narrow set of examples.

Highlighting the first example you asked:
> Does this means that links must have a box around them and cannot use an underline for highlighting? Focus Appearance (Minimum) 2.4.11 has an example of an underline to indicate the selected location. 
>
> Can the group provide guidance on when you can use an underline and when you must use an outline? Requiring consistency would likely be very helpful.

The requirement is for a certain surface area that is at least as much as a 2px outline. You could use an underline, but it would have to be very thick. The understanding document for the AA version includes many examples, but providing best-practice guidance goes beyond the scope of the understanding document. That sort of guidance goes in the [WAI turorials](https://www.w3.org/WAI/tips/designing/#ensure-that-interactive-elements-are-easy-to-identify).

We did consider requiring a consistent approach to focus indicators, but there are instances where it can help to vary the focus indicator, e.g. with different components on the same page.

> As we raised in 2.4.11 Focus Appearance (Minimum), we are similarly concerned about small components, like a radio button, whose focus indicator is hugging the visible boundary it may be too small for some users with low vision to see.
> 
> Can a minimum size for the focus indicator be provided? Will similar details from 2.4.11 be added here as well?

We did look at a baseline or fixed-size approach, but it is difficult to determine what that would be without considering the size of the component. A 2px solid border is significantly better than the current default. The following shows the default firefox dotted outline compared to a solid border.

<img src="https://user-images.githubusercontent.com/216630/107154379-8f815600-696a-11eb-80a9-0cbabb178163.png" alt="two radio buttons, the first has a thin dotted outline, the second a thicker solid outline.">

If you have a suggestion for how that would work, we'd happily consider that.

> Can the group please share what influenced the decision to recommend a two-color focus indicator as a sufficient technique? 

It was aiming to provide one focus style that could work across any background or foreground styles, so designers/developers on smaller teams could implement one style that worked anywhere. Obviously teams which use component libraries or have more resources can customise focus styles per component, but smaller teams/individuals may need a more efficient solution.

**Comment by alastc** [comment]
The group approved the above response.
